File: src/097. Large non-Mersenne prime/097.go
package main

import (
	"fmt"
	"math/big"
)

func main() {
	var m big.Int
	m.Exp(big.NewInt(10), big.NewInt(10), nil)

	var result big.Int
	result.Exp(big.NewInt(2), big.NewInt(7830457), &m)
	result.Mul(big.NewInt(28433), &result)
	result.Add(&result, big.NewInt(1))
	result.Mod(&result, &m)

	fmt.Println(result.String())
}

File: src/097. Large non-Mersenne prime/097.java
import java.math.BigInteger;

public class Main {
    public static void main(String[] args) {
        final BigInteger ten2Ten = BigInteger.valueOf(10).pow(10);
        BigInteger powerTwo = BigInteger.valueOf(2).modPow(BigInteger.valueOf(7830457), ten2Ten);
        BigInteger prime = powerTwo.multiply(BigInteger.valueOf(28433)).add(BigInteger.ONE);
        BigInteger result = prime.mod(ten2Ten);
        System.out.println(result);
    }
}

